store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -161,7 +219,14 @@ public CarbonReaderBuilder withRowRecordReader() {
     }
     CarbonTable table;
     // now always infer schema. TODO:Refactor in next version.
-    table = CarbonTable.buildTable(tablePath, tableName, hadoopConf);
+    if (null == this.fileLists && null == tablePath) {
+      throw new IllegalArgumentException("Please set table path first.");
+    }
+    if (null != this.fileLists) {
+      table = CarbonTable.buildTable(this.fileLists.get(0).toString(), tableName, hadoopConf, true);
```

Comment:
Need to set folder path instead of file path as table path ?

Reply:
compatible